test/forms/mappings/LocalDateFormatterSpec.scala

```diff
@@ -327,19 +342,156 @@ class LocalDateFormatterSpec extends AnyFreeSpec with Matchers with ScalaCheckPr
+        FormError("value", "error.invalid.day", List(30, "day"))
+      )
+    }
+
+    "when date is 32nd of a month with only 28 days" in {
+
+      val data = Map(
+        "value.day"   -> "32",
+        "value.month" -> "02",
+        "value.year"  -> "2025"
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.day", List(28, "day"))
+      )
+    }
+
+    "when date is 31st of a month with only 28 days in an invalid year" in {
+
+      val data = Map(
+        "value.day"   -> "31",
+        "value.month" -> "02",
+        "value.year"  -> "foo"
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.multiple", List("day", "year"))
+      )
+    }
+
+    "when date is 31st of a month with max 29 days in an unknown year" in {
+
+      val data = Map(
+        "value.day"   -> "31",
+        "value.month" -> "02",
+        "value.year"  -> ""
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.day", List(29, "day")),
+        FormError("value", "error.required.year", List("year"))
+      )
+    }
+
+    "when date contains a month not between 1 and 12" in {
+
+      val data = Map(
+        "value.day"   -> "15",
+        "value.month" -> "13",
+        "value.year"  -> "2025"
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.month", Seq("month"))
+      )
+    }
+
+    "when date contains an invalid day and month" in {
+
+      val data = Map(
+        "value.day"   -> "32",
+        "value.month" -> "13",
+        "value.year"  -> "2025"
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.multiple", Seq("day", "month"))
+      )
+    }
+
+    "when date contains an invalid day, month and year" in {
+
+      val data = Map(
+        "value.day"   -> "foo",
+        "value.month" -> "foo",
+        "value.year"  -> "foo"
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.all", Seq("day", "month", "year"))
+      )
+    }
+
+    "when date contains an invalid year" in {
+
+      val data = Map(
+        "value.day"   -> "12",
+        "value.month" -> "12",
+        "value.year"  -> "foo"
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.year", Seq("year"))
+      )
+    }
+
+    "when date contains a year that cannot be parsed as an integer" in {
+
+      val data = Map(
+        "value.day"   -> "12",
+        "value.month" -> "12",
+        "value.year"  -> "99999999999"
+      )
+
+      val result = form.bind(data)
+
+      result.errors mustEqual Seq(
+        FormError("value", "error.invalid.year", Seq("year"))
```

Comment:
Shouldn't this be dateGoodsUnloaded.error.max.date = The date must be today or in the past